explorer/interpreter/type_checker.cpp

```diff
@@ -662,6 +663,21 @@ auto TypeChecker::ArgumentDeduction(
       if (std::find(bindings_to_deduce.begin(), bindings_to_deduce.end(),
                     &var_type.binding()) != bindings_to_deduce.end()) {
         auto [it, success] = deduced.insert({&var_type.binding(), arg});
+
+        const Value* substituted_param_type = Substitute(deduced, param);
+        const auto& binding = cast<VariableType>(*param).binding();
+        if (binding.has_static_type()) {
+          if (!IsTypeOfType(&binding.static_type())) {
+            if (!IsImplicitlyConvertible(substituted_param_type,
+                                         &binding.static_type(), impl_scope,
+                                         false)) {
```

Comment:
```suggestion
            if (!IsImplicitlyConvertible(substituted_param_type,
                                         arg, impl_scope, false)) {
```
It seems to me that we should be checking that the argument converts to the parameter type here, not that the parameter converts to its own substituted type. Or am I missing something?